src/node_i18n.cc

```diff
@@ -62,6 +72,300 @@ bool flag_icu_data_dir = false;
 
 namespace i18n {
 
+const size_t kStorageSize = 1024;
+
+Local<Value> ICUException(Environment* env,
+                          UErrorCode status,
+                          const char* msg) {
+  if (msg == nullptr || msg[0] == '\0')
+    msg = "Unspecified ICU Exception";
+  char buf[kStorageSize];
+  int len = snprintf(buf, sizeof(buf), "%s [%s]", msg, u_errorName(status));
+
+  Local<Value> e =
+      Exception::Error(String::NewFromUtf8(env->isolate(),
+                       buf, v8::NewStringType::kNormal,
+                       len).ToLocalChecked());
+  if (e.IsEmpty())
+    return Local<Value>();
+  Local<Object> obj = e->ToObject(env->isolate());
+  obj->Set(env->errno_string(), Integer::New(env->isolate(), status));
+  obj->Set(env->code_string(),
+           String::NewFromUtf8(env->isolate(),
+                               u_errorName(status), v8::NewStringType::kNormal)
+                                 .ToLocalChecked());
```

Comment:
The new `v8::Maybe<T>` API for `v8::Object::Set()` is annoying and ugly, but if if we're going to use some of the new API might as well use all of it.